driver/config/provider_viper.go

```diff
@@ -82,22 +84,31 @@ const (
 	ViperKeyHasherArgon2ConfigParallelism                           = "hashers.argon2.parallelism"
 	ViperKeyHasherArgon2ConfigSaltLength                            = "hashers.argon2.salt_length"
 	ViperKeyHasherArgon2ConfigKeyLength                             = "hashers.argon2.key_length"
+	ViperKeyHasherArgon2ConfigMinimalDuration                       = "hashers.argon2.minimal_duration"
+	ViperKeyHasherArgon2ConfigExpectedDeviation                     = "hashers.argon2.expected_deviation"
+	ViperKeyHasherArgon2ConfigDedicatedMemory                       = "hashers.argon2.dedicated_memory"
 	ViperKeyPasswordMaxBreaches                                     = "password.max_breaches"
 	ViperKeyIgnoreNetworkErrors                                     = "password.ignore_network_errors"
 	ViperKeyVersion                                                 = "version"
-	Argon2DefaultMemory                                      uint32 = 4 * 1024 * 1024
-	Argon2DefaultIterations                                  uint32 = 4
+	Argon2DefaultMemory                                             = uint32(512 * bytesize.MB / bytesize.KB)
+	Argon2DefaultIterations                                  uint32 = 1
 	Argon2DefaultSaltLength                                  uint32 = 16
 	Argon2DefaultKeyLength                                   uint32 = 32
+	Argon2DefaultDuration                                           = 500 * time.Millisecond
+	Argon2DefaultDeviation                                          = 500 * time.Millisecond
+	Argon2DefaultDedicatedMemory                                    = 4 * bytesize.GB
```

Comment:
I would set this to 1:

```suggestion
	Argon2DefaultDedicatedMemory                                    = 1 * bytesize.GB
```

Reply:
This parameter is the point where we will queue hashing operations and warn in the logs when it is reached, not a hard limit. Same considerations as above.

Reply:
I see!